master/internal/config/config.go

```diff
@@ -236,67 +236,42 @@ type ObservabilityConfig struct {
 	EnablePrometheus bool `json:"enable_prometheus"`
 }
 
-func readPreemptionFromScheduler(conf *resourcemanagers.SchedulerConfig) *bool {
-	if conf == nil || conf.Priority == nil {
-		return nil
-	}
-	return &conf.Priority.Preemption
-}
-
 func readPriorityFromScheduler(conf *resourcemanagers.SchedulerConfig) *int {
 	if conf == nil || conf.Priority == nil {
 		return nil
 	}
 	return conf.Priority.DefaultPriority
 }
 
-// ReadPreemptionStatus resolves the desired preemption status for a job.
-func ReadPreemptionStatus(rpName string, jobConf interface{}) bool {
+// ReadRMPreemptionStatus resolves the preemption status for a resource manager.
+func ReadRMPreemptionStatus(rpName string) bool {
 	config := GetMasterConfig()
```

Comment:
didn't get to look at the previous pr, this master config singleton thing is pretty strange to me, why did we need this? things like this have a tendency to cause weird bugs/makes the system really hard to break up / untangle later. also it seems the `sync.Once` is just so GetMasterConfig works in tests (it is always set otherwise)? they should probably just set a default config, since this could cause a really bad bug if it ever gets called wrong?

Reply:
should we chat about this over at the [the singleton config pr](https://github.com/determined-ai/determined/pull/3277)
I might have not gotten the second part correctly but the sync.Once is to ensure the singleton isn't initialized twice if the correct interfaces are used.

Reply:
yeah i can move it over there